app/Http/Controllers/AuthController.php

```diff
@@ -52,26 +76,134 @@ public function login(Request $request)
     public function register(Request $request)
     {
         $this->validate($request, [
-            'email' => 'required',
-            'password' => 'required',
-            'name' => 'string'
+            'email' => 'required|email',
+            'password' => 'required|min:9',
         ]);
 
         $emailInUse = User::where('email', $request->input('email'))->exists();
 
-        if ($emailInUse) {
-            return response()->json(['message' => 'Email already in use!'], 400);
+        if (!$emailInUse) {
+            $user = new User;
+            $user->email = $request->input('email');
+            $user->password = Hash::make($request->input('password'));
+            $user->save();
+
+            $verifyUser = VerifyUser::create([
+                'user_id' => $user->id,
+                'token' => str_random(60)
+            ]);
+
+            Mail::to($user->email)->send(new VerifyMail($user));
+        }
+        return response()->json([
+            'status' => 'success',
+            'message' => 'Check email to verify account'
+        ]);
+    }
+
+    /**
+    * @param String $token
+    *
+    * @return Response
+    */
+    public function verify($token)
+    {
+        $verifyUser = VerifyUser::where('token', $token)->first();
+        if(isset($verifyUser) ){
+            $verifyUser->user->verified = 1;
+            $verifyUser->user->save();
+            $verifyUser->delete();
+            return response()->json([
+                'status' => 'success',
+                'message' => 'verified'
+            ]);
+        }
+        else {
+            return response()->json([
+                'status' => 'invalid.token',
+                'message' => 'Invalid verification token'
+            ], 400);
+        }
+    }
+
+    /**
+    * @param Request $request
+    *
+    * @return Response
+    */
+
+    public function refresh(Request $request)
+    {
+        if ( $token = Auth::refresh($request) ) {
+            return response([
+                'status' => 'success'
+            ])
+            ->header('Authorization', $token);
         }
+        else {
```

Comment:
no return needed

Reply:
Without returning an error message, how could the user know that the refresh token is invalid and that he needs to login again?

Reply:
sry meant no else needed